### PR TITLE
feat: style SessionUserMenu

### LINKED ISSUE #147: Add Sign In/Sign Up/User Menu to Nav Bar 🎨
Add signin/signup to Nav Bar for anonymous users.  When signed in, show the user menu.

### PR DESCRIPTION
**Pull request by chriswhong**
- styles the sign up and log in buttons for anonymous users
- styles the user menu icon for logged in users
- updates the user menu items
- changes navbar link styles
- adds a default size to `QriLogo` which was not displaying in the navbar due to a missing `size prop`
- adds hover styles to `ButtonGroup` (top my datasets and dashboard buttons)
- animates the caret that is used to expand the dataset nav

Closes #147

### CODE REVIEW

File: frontend/src/chrome/DropdownMenu.tsx [hunk 1]
```diff
@@ -19,10 +20,16 @@ export interface DropDownMenuItem {
...
+  className?: string
 }
 
 // itemProps will be passed into the onClick handler for each item in the dropdown
-const DropdownMenu: React.FC<DropdownMenuProps> = ({ children, items, alignLeft=false }) => {
+const DropdownMenu: React.FC<DropdownMenuProps> = ({
+  children,
+  items,
+  alignLeft=false,
+  className
```
Comment (b5) [inline frontend/src/chrome/DropdownMenu.tsx]: give `className` a default string value

File: frontend/src/features/navbar/SessionUserMenu.tsx [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
-import Icon from '../../chrome/Icon'
 import DropdownMenu from '../../chrome/DropdownMenu'
 import { AnonUser, selectSessionUser } from '../session/state/sessionState'
 import { logOut } from '../session/state/sessionActions'
 import { showModal } from '../app/state/appActions'
 import { ModalType } from '../app/state/appState'
 import Button from '../../chrome/Button'
+import ExternalLink from '../../chrome/ExternalLink'
+
+const navbarLinkClassNames = 'text-qriblue font-medium hover:text-qriblue-800 hover:cursor-pointer transition-all duration-100'
```
Comment (b5) [inline frontend/src/features/navbar/SessionUserMenu.tsx:12]: any reason this is being pulled up into a constant? Looks like the string can be kept inline

**Review by b5 — APPROVED** [review]
two nits, then looks great

File: frontend/src/features/navbar/SessionUserMenu.tsx (hunk 2 above)
Comment (chriswhong) [reply frontend/src/features/navbar/SessionUserMenu.tsx:12]: Just because it is used in two places in this component.

**Pull request merged**